docs/sdk/src/reference_docs/defensive_programming.rs

```diff
@@ -0,0 +1,618 @@
+    increase_balance_match
+)]
+//!
+//! This is generally a useful convention for handling not only checked types, but most types that
+//! return `Option<T>`.
+//!
+//! #### Handling via Result - Less Verbose
+//!
+//! In the Polkadot SDK codebase, you may see checked operations being handled as a [`Result`] via
+//! `ok_or`. This is a less verbose way of expressing the above.  Which to use often boils down to
+//! the developer's preference:
+#![doc = docify::embed!(
+    "./src/reference_docs/defensive_programming.rs",
+    increase_balance_result
+)]
+//!
+//! ### Saturating Operations
+//!
+//! Saturating a number limits it to the type's upper or lower bound, no matter the integer would
+//! overflow in runtime. For example, adding to `u32::MAX` would simply limit itself to `u32::MAX`:
+#![doc = docify::embed!(
+    "./src/reference_docs/defensive_programming.rs",
+    saturated_add_example
+)]
+//!
+//! Saturating calculations can be used if one is very sure that something won't overflow, but wants
+//! to avoid introducing the notion of any potential-panic or wrapping behavior.
+//!
+//! There is also a series of defensive alternatives via
+//! [`DefensiveSaturating`](frame::traits::DefensiveSaturating), which introduces the same behavior
+//! of the [`Defensive`](frame::traits::Defensive) trait, only with saturating, mathematical
+//! operations:
+#![doc = docify::embed!(
+    "./src/reference_docs/defensive_programming.rs",
+    saturated_defensive_example
+)]
+//!
+//! ### Mathematical Operations in Substrate Development - Further Context
+//!
+//! As a recap, we covered the following concepts:
+//!
+//! 1. **Checked** operations - using [`Option`] or [`Result`]
+//! 2. **Saturating** operations - limited to the lower and upper bounds of a number type
+//! 3. **Wrapped** operations (the default) - wrap around to above or below the bounds of a type
+//!
+//!
+//! Known scenarios that could be fallible should be avoided: i.e., avoiding the possibility of
+//! dividing/modulo by zero at any point should be mitigated. One should be, instead, opting for a
+//! `checked_*` method in order to introduce safe arithmetic in their code.
+//!
+//! #### The problem with 'default' wrapped operations
+//!
+//! **Wrapped operations** cause the overflow to wrap around to either the maximum or minimum of
+//! that type. Imagine this in the context of a blockchain, where there are balances, voting
+//! counters, nonces for transactions, and other aspects of a blockchain.
+//!
+//! Some of these mechanisms can be more critical than others. It's for this reason that we may
+//! consider some other ways of dealing with runtime arithmetic, such as saturated or checked
+//! operations, that won't carry these potential consequences.
+//!
+//! While it may seem trivial, choosing how to handle numbers is quite important. As a thought
+//! exercise, here are some scenarios of which will shed more light on when to use which.
```

Comment:
Removing redundant info that was already presented and emphasized before.

```suggestion
//!
//! While it may seem trivial, choosing how to handle numbers is quite important. As a thought
//! exercise, here are some scenarios of which will shed more light on when to use which.
```